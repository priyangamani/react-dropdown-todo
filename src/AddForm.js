import React, { useState } from "react";
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import styles from './styles';
import "./App.css";
import Alert from '@material-ui/lab/Alert';

function AddForm({ addValue }) {
  const classes = styles();
  const [value, setValue] = useState("");
  const [addItem, setAddItem] = useState(false);
  const handleSubmit = e => {
    e.preventDefault();
    if (!value) return;
    addValue(value);
    setValue("");
    setAddItem(true);
  };

  const handleChange = prop => event => {
    setAddItem(false);
    setValue(event.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <TextField
        id="outlined-adornment-password"
        variant="outlined"
        type="text"
        label="Add Country"
        value={value}
        onChange={handleChange("country")}
        className={classes.inputContainerWidth}
      />
      <Button className={[classes.buttonWidth, classes.buttonContainer]} variant="contained" color="primary" onClick={handleSubmit}>
        Add & Select
      </Button>
      {addItem && <Alert severity="success">
         Added Successfully â€” <strong>check it out!</strong>
      </Alert>}
    </form>
  );
}

export default AddForm;