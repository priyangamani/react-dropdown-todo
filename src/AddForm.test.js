import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react'
import AddForm from './AddForm';

test('renders addForm', () => {
  const tree = render(<AddForm />);
  fireEvent.click(screen.getByText('Add & Select'))
  expect(tree).toMatchSnapshot();
});
