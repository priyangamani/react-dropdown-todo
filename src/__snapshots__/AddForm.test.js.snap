// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders addForm 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root makeStyles-inputContainerWidth-1"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
            data-shrink="false"
            for="outlined-adornment-password"
            id="outlined-adornment-password-label"
          >
            Add Country
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input"
              id="outlined-adornment-password"
              type="text"
              value=""
            />
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="PrivateNotchedOutline-legendLabelled-7"
              >
                <span>
                  Add Country
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonWidth-2 makeStyles-buttonContainer-4 MuiButton-containedPrimary"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Add & Select
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form>
      <div
        class="MuiFormControl-root MuiTextField-root makeStyles-inputContainerWidth-1"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
          for="outlined-adornment-password"
          id="outlined-adornment-password-label"
        >
          Add Country
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input"
            id="outlined-adornment-password"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-7"
            >
              <span>
                Add Country
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-buttonWidth-2 makeStyles-buttonContainer-4 MuiButton-containedPrimary"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Add & Select
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
